src/dnslib/dns.py

```diff
@@ -0,0 +1,3341 @@
+        41: "OPT",
+        42: "APL",
+        43: "DS",
+        44: "SSHFP",
+        45: "IPSECKEY",
+        46: "RRSIG",
+        47: "NSEC",
+        48: "DNSKEY",
+        49: "DHCID",
+        50: "NSEC3",
+        51: "NSEC3PARAM",
+        52: "TLSA",
+        53: "HIP",
+        55: "HIP",
+        59: "CDS",
+        60: "CDNSKEY",
+        61: "OPENPGPKEY",
+        62: "CSYNC",
+        63: "ZONEMD",
+        64: "SVCB",
+        65: "HTTPS",
+        99: "SPF",
+        108: "EUI48",
+        109: "EUI64",
+        249: "TKEY",
+        250: "TSIG",
+        251: "IXFR",
+        252: "AXFR",
+        255: "ANY",
+        256: "URI",
+        257: "CAA",
+        32768: "TA",
+        32769: "DLV",
+    },
+    unknown_qtype,
+)
+"""QTYPE `Bimap` - allows bi-directional mapping between integer and text representations"""
+
+CLASS = Bimap("CLASS", {1: "IN", 2: "CS", 3: "CH", 4: "Hesiod", 254: "None", 255: "*"}, DNSError)
+"""CLASS `Bimap` - allows bi-directional mapping between integer and text representations"""
+
+QR = Bimap("QR", {0: "QUERY", 1: "RESPONSE"}, DNSError)
+"""QR `Bimap` - allows bi-directional mapping between integer and text representations"""
+
+RCODE = Bimap(
+    "RCODE",
+    {
+        0: "NOERROR",
+        1: "FORMERR",
+        2: "SERVFAIL",
+        3: "NXDOMAIN",
+        4: "NOTIMP",
+        5: "REFUSED",
+        6: "YXDOMAIN",
+        7: "YXRRSET",
+        8: "NXRRSET",
+        9: "NOTAUTH",
+        10: "NOTZONE",
+    },
+    DNSError,
+)
+"""RCODE `Bimap` - allows bi-directional mapping between integer and text representations"""
+
+OPCODE = Bimap("OPCODE", {0: "QUERY", 1: "IQUERY", 2: "STATUS", 4: "NOTIFY", 5: "UPDATE"}, DNSError)
+"""OPCODE `Bimap` - allows bi-directional mapping between integer and text representations"""
+
+
+def create_label(label: str, origin: DNSLabelCreateTypes = None) -> DNSLabel:
+    """Create a DNSLabel from a string
+
+    Args:
+        label:
+        origin: base of label if label is not connected to root (`.`).
+
+    Changed in 1.0: renamed from `label` to `create_label` (avoids name collisions within classes)
+    """
+    # TODO: Should this be a classmethod on DNSLabel? Could it just be apart of init?
+    if label.endswith("."):
+        return DNSLabel(label)
+    if not isinstance(origin, DNSLabel):
+        origin = DNSLabel(origin)
+    return origin.add(label)
+
+
+def create_label_property(attr: str = "label"):
+    """Property creator for DNSLabel properties
+
+    Args:
+        attr: name of attribute
+    """
+    obj_attr = f"_{attr}"
+
+    def getter(self) -> DNSLabel:
+        return getattr(self, obj_attr)
+
+    def setter(self, label: DNSLabelCreateTypes) -> None:
+        if not isinstance(label, DNSLabel):
+            label = DNSLabel(label)
+        setattr(self, obj_attr, label)
+        return
+
+    return property(getter, setter)
+
+
+# TODO: This should potentially be renamed to the RFC1035 name of `DNSMessage`
```

Comment:
Per code comment.

If we think we want to change the name now would be a good time to do it since there are a lot of breaking changes in this version